homeassistant/components/nordpool/sensor.py

```diff
@@ -0,0 +1,128 @@
+"""Support for Nordpool electrical prices sensors."""
+import logging
+
+from datetime import datetime
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (ATTR_ATTRIBUTION,
+                                 CONF_NAME,
+                                 CONF_CURRENCY,
+                                 CONF_REGION)
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['nordpool==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:cash-usd'
+ATTRIB = "For details, see https://nordpoolgroup.com/"
+
+CURRENCY_FRACTION = {
+    'DKK': 'Øre',
+    'EUR': 'Cent',
+    'NOK': 'Øre',
+    'SEK': 'Øre'
+}
+
+VALID_REGION = ['DK1',
+                'DK2',
+                'EE',
+                'FI',
+                'LT',
+                'LV',
+                'Oslo',
+                'Kr.sand',
+                'Bergen',
+                'Molde',
+                'Tr.heim',
+                'Tromsø',
+                'SE1',
+                'SE2',
+                'SE3',
+                'SE4',
+                'SYS']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_CURRENCY): vol.All(cv.string, vol.In(CURRENCY_FRACTION)),
+    vol.Required(CONF_REGION): vol.All(cv.string, vol.In(VALID_REGION)),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Nordpool sensor platform."""
+    sens_name = config[CONF_NAME]
+    sens_currency = config[CONF_CURRENCY]
+    sens_region = config[CONF_REGION]
+
+    sens_nordpool = NordpoolAPI(sens_name, sens_currency, sens_region)
+    add_entities([sens_nordpool], True)
+
+
+class NordpoolAPI(Entity):
+    """Initiate communications to Nordpool API."""
+
+    def __init__(self, sens_name, sens_currency, sens_region):
+        """Init new sensor."""
+        self._name = sens_name
+        self._unit_of_measurement = CURRENCY_FRACTION[sens_currency]
+        self._region = sens_region
+        self._icon = ICON
+        self._state = None
+        self._currency = sens_currency
+        self._day = None
+        self._data = None
+        self._device_state_attributes = {}
+
+    def update(self):
+        """Update sensor values."""
+        from nordpool import elspot
+
+        get_date = datetime.now()
+
+        if get_date.day != self._day:
```

Comment:
If we don't update the state via a new api call, we should just return after this check if it fails.